stan/math/gpu/kernels/lower_tri_inverse_step1.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef STAN_MATH_GPU_KERNELS_LOWER_TRI_INVERSE_STEP1_HPP
+#define STAN_MATH_GPU_KERNELS_LOWER_TRI_INVERSE_STEP1_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/gpu/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+// \cond
+const char* lower_tri_inverse_step1_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * Calculates the inverse with no blocking
+     *
+     * @param[in,out] A The input matrix.
+     * @param[in, out] V A temporary matrix for storing partial calculations.
+     * @param rows The number of rows for A.
+     * @note Code is a <code>const char*</code> held in
+     * <code>lower_tri_inverse_step1_kernel_code.</code>
+     *  Used in math/gpu/lower_tri_inverse.hpp.
+     *  This kernel uses the helper macros available in helpers.cl.
+     */
+    __kernel void lower_tri_inverse_step1(__global double* A,
+                                          __global double* V, int rows) {
+      int index = get_local_id(0);
+      int group = get_group_id(0);
+      int block_size = get_local_size(0);
+      int offset = group * block_size;
+      int i = index;
+      for (int j = 0; j < block_size; j++) {
+        if (i == j) {
+          V[group * block_size * block_size + j * block_size + i] = 1.0;
+        } else {
+          V[group * block_size * block_size + j * block_size + i] = 0.0;
+        }
+      }
+      barrier(CLK_LOCAL_MEM_FENCE);
+      for (int k = 0; k < block_size; k++) {
+        double factor = A(offset + k, offset + k);
+        int j = index;
```

Comment:
No, this was not cleaned up... My bad. I was trying a few different versions to see what would work better. Sorry :/

Reply:
No worries! I think something like this might make sense if, as you mentioned for the M, N, K names, i, j, and k are also always used to iterate over those. I think it probably makes sense for us to think of our target audience as people already steeped in linear algebra from the sources that use those naming conventions, if those are the standard. If we're coming up with our own names I would love to be able to tell by looking at it which dimension on which matrix it should be iterating along.